Example/Example ObjC/ScreenAPIViewController.m

```diff
@@ -48,134 +45,48 @@ - (IBAction)easyLaunchGiniVision:(id)sender {
-    NSLog(@"Screen API canceled review");
-    
-    // Cancel analysis process to avoid unnecessary network calls.
-    [self cancelAnalysis];
-}
-
-- (void)didShowAnalysis:(id<AnalysisDelegate>)analysisDelegate {
-    NSLog(@"Screen API started analysis screen");
-    
-    _analysisDelegate = analysisDelegate;
-    
-    // The analysis screen is where the user should be confronted with any errors occuring during the analysis process.
-    // Show any errors that occured while the user was still reviewing the image here.
-    // Make sure to only show errors relevant to the user.
-    [self showErrorMessage];
-}
-
-- (void)didCancelAnalysis {
-    NSLog(@"Screen API canceled analysis");
-    
-    _analysisDelegate = nil;
-    
-    // Cancel analysis process to avoid unnecessary network calls.
-    [self cancelAnalysis];
-}
-
-// MARK: Handle analysis of document
-- (void)analyzeDocumentWithImageData:(NSData *)data {
-    [self cancelAnalysis];
-    _imageData = data;
-    [[AnalysisManager sharedManager] analyzeDocumentWithImageData:data cancelationToken:[CancelationToken new] andCompletion:^(NSDictionary *result, GINIDocument * document, NSError *error) {
-        if (error) {
-            self.errorMessage = @"Es ist ein Fehler aufgetreten. Wiederholen";
-        } else if (result && document) {
-            self.document = document;
-            self.result = result;
-        } else {
-            self.errorMessage = @"Ein unbekannter Fehler ist aufgetreten. Wiederholen";
-        }
-    }];
-}
-
-- (void)cancelAnalysis {
-    [[AnalysisManager sharedManager] cancelAnalysis];
-    _result = nil;
-    _document = nil;
-    _errorMessage = nil;
-    _imageData = nil;
-}
-
-// MARK: Handle results from analysis process
-- (void)setErrorMessage:(NSString *)errorMessage {
-    _errorMessage = errorMessage;
-    if (_errorMessage) {
-        [self showErrorMessage];
-    }
-}
-
-- (void)setResult:(NSDictionary *)result {
-    _result = result;
-    if (_result && _document) {
-        [self showResults];
-    }
-}
-
-- (void)showErrorMessage {
-    if (_errorMessage && _imageData && _analysisDelegate) {
-        [_analysisDelegate displayErrorWithMessage:_errorMessage andAction:^{
-            [self analyzeDocumentWithImageData: _imageData];
-        }];
-    }
-}
-
-- (void)showResults {
-    if (_analysisDelegate) {
-        _analysisDelegate = nil;
-        [self presentResults];
-    }
+- (void)giniVision:(NSArray<id<GiniVisionDocument>> *)documents
+analysisDidFinishWithResults:(NSDictionary<NSString *,GINIExtraction *> *)results
+      sendFeedback:(void (^)(NSDictionary<NSString *,GINIExtraction *> * _Nonnull))sendFeedback {
+    _result = results;
+    [self presentResultsWithSendFeedbackBlock:sendFeedback];
 }
 
-- (void)presentResults {
+- (void)presentResultsWithSendFeedbackBlock:(SendFeedbackBlock)sendFeedbackBlock {
     NSArray *payFive = @[@"paymentReference", @"iban", @"bic", @"amountToPay", @"paymentRecipient"];
```

Comment:
Do we still need to check for the pay5? I vaguely remember that we decided to work even if some of them are missing. Moreover, the BIC (and even the recipient for that matter) is not mandatory to make a transfer.

Reply:
It is merely an example, they may choose to take only the IBAN. But I guess that adding a comment can clarify it.